app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1743,16 +1748,103 @@ impl GraphEditorModel {
+        for node in nodes.values() {
+            let node_bounding_box = node.frp.bounding_box.value();
+            let distance_squared = node_bounding_box.squared_distance_to_point(point);
+            if distance_squared < min_distance_squared {
+                min_distance_squared = distance_squared;
+                nearest_node = Some(node.clone_ref());
+            }
+        }
+        nearest_node
+    }
+
+    /// Calculates a position for a new node, aligning it to the specified reference node if the
+    /// proposed position is close enough to it.
+    ///
+    /// A point is close enough to a node if it is located in an alignment area around a node,
+    /// defined in the current theme ([`theme::graph_editor::alignment_area_around_node`]).
+    pub fn new_node_position_aligned_if_close_to_node(
+        &self,
+        proposed_position: Vector2,
+        alignment_node: Option<Node>,
+    ) -> Vector2 {
+        let alignment_node = alignment_node.filter(|node| {
+            let node_bounding_box = node.frp.bounding_box.value();
+            let styles = &self.styles_frp;
+            use theme::graph_editor::alignment_area_around_node as alignment_area_style_path;
+            let distance_from_left_node_edge_to_alignment_area_left_edge =
+                styles.get_number_or(alignment_area_style_path::distance_to_the_left_of_node, 0.0);
+            let alignment_area_min_x = node_bounding_box.left()
+                - distance_from_left_node_edge_to_alignment_area_left_edge.value();
+            let distance_from_right_node_edge_to_alignment_area_right_edge =
+                styles.get_number_or(alignment_area_style_path::distance_to_the_right_of_node, 0.0);
+            let alignment_area_max_x = node_bounding_box.right()
+                + distance_from_right_node_edge_to_alignment_area_right_edge.value();
+            let distance_from_top_node_edge_to_alignment_area_top_edge =
+                styles.get_number_or(alignment_area_style_path::distance_above_node, 0.0);
+            let alignment_area_max_y = node_bounding_box.top()
+                + distance_from_top_node_edge_to_alignment_area_top_edge.value();
+            let distance_from_bottom_node_edge_to_alignment_area_bottom_edge =
+                styles.get_number_or(alignment_area_style_path::distance_below_node, 0.0);
+            let alignment_area_min_y = node_bounding_box.bottom()
+                - distance_from_bottom_node_edge_to_alignment_area_bottom_edge.value();
+            let alignment_area = selection::BoundingBox::from_corners(
+                Vector2(alignment_area_min_x, alignment_area_min_y),
+                Vector2(alignment_area_max_x, alignment_area_max_y),
+            );
+            alignment_area.contains(proposed_position)
+        });
+        match alignment_node {
+            Some(node) => self.find_free_place_under(node.id()),
+            None => proposed_position,
+        }
```

Comment:
This code is a lot more understnadable than it was, but now it uses super long names everywhere (still, it's better this way rather than the last time when the names were not understandable). As I explained during the last call, there are two approaches here – you can either use longer names, or a better documentation, which explains the ideas better and clearly shows where short names are used. For example, the above code can be rewritten to this:

```rust


     /// Calculates a position for a new node, aligning it to the specified reference node if the
     /// proposed position is close enough to it.
     ///
     /// A point is close enough to a node if it is located in an alignment area around a node,
     /// defined in the current theme ([`theme::graph_editor::alignment_area_around_node`]).
     ///
     /// In the picture below, the solid border represents the node, while the dashed-border 
     /// represents the alignment area. The captions are used as the variables in the code.
     /// ┌┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┐
     /// ┆               ▲               ┆
     /// ┆           top │               ┆
     /// ┆               ▼               ┆
     /// ┆ left  ┌───────────────┐ right ┆
     /// ┆ ◀───▶ └───────────────┘ ◀───▶ ┆
     /// ┆               ▲               ┆
     /// ┆        bottom │               ┆
     /// ┆               ▼               ┆
     /// └┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┘
     pub fn new_node_position_aligned_if_close_to_node(
         &self,
         proposed_position: Vector2,
         alignment_node: Option<Node>,
     ) -> Vector2 {
         let alignment_node = alignment_node.filter(|node| {
             use theme::graph_editor::alignment_area_around_node as alignment_area_style;
             let node_bounding_box = node.frp.bounding_box.value();
             let styles = &self.styles_frp;
             let left = styles.get_number_or(alignment_area_style::distance_to_the_left_of_node, 0.0);
             let alignment_area_min_x = node_bounding_box.left() - left.value();
             let right = styles.get_number_or(alignment_area_style::distance_to_the_right_of_node, 0.0);
             let alignment_area_max_x = node_bounding_box.right() + right.value();
             let top = styles.get_number_or(alignment_area_style::distance_above_node, 0.0);
             let alignment_area_max_y = node_bounding_box.top() + top.value();
             let bottom = styles.get_number_or(alignment_area_style::distance_below_node, 0.0);
             let alignment_area_min_y = node_bounding_box.bottom() - bottom.value();
             let alignment_area = selection::BoundingBox::from_corners(
                 Vector2(alignment_area_min_x, alignment_area_min_y),
                 Vector2(alignment_area_max_x, alignment_area_max_y),
             );
             alignment_area.contains(proposed_position)
         });
         match alignment_node {
             Some(node) => self.find_free_place_under(node.id()),
             None => proposed_position,
         }
     }
```